pkg/ruler/ruler.go

```diff
@@ -409,12 +394,46 @@ func (r *Ruler) loadRules(ctx context.Context) {
 		if _, exists := configs[user]; !exists {
 			go mngr.Stop()
 			delete(r.userManagers, user)
-			level.Info(util.Logger).Log("msg", "deleting rule manager", "user", user)
+			level.Info(r.logger).Log("msg", "deleting rule manager", "user", user)
 		}
 	}
 
 }
 
+// syncManager maps the rule files to disk, detects any changes and will create/update the
+// the users Prometheus Rules Manager
+func (r *Ruler) syncManager(ctx native_ctx.Context, user string, groups store.RuleGroupList) {
+	r.userManagerMtx.Lock()
+	defer r.userManagerMtx.Unlock()
+
+	// Map the files to disk and return the file names to be passed to the users manager
+	update, files, err := r.mapper.MapRules(user, groups.Formatted())
+	if err != nil {
+		level.Error(r.logger).Log("msg", "unable to map rule files", "user", user, "err", err)
+		return
+	}
+
+	if update {
+		level.Info(r.logger).Log("msg", "updating rules", "user", "user")
```

Comment:
Info or debug level? Up to you, just wanted to raise it, cause may be a bit verbose for an Info level.

Reply:
I think an update would be a relatively rare event? Since we have a metric I'm ok with lowering it to debug though.